tests/conftest.py

```diff
@@ -1577,6 +1580,15 @@ async def recorder_mock(
     return await async_setup_recorder_instance(hass, recorder_config)
 
 
+@pytest.fixture
+def setup_recorder_before_hass() -> RecorderInstanceGenerator | None:
+    """Mock the recorder.
+
+    Override or parametrize this fixture with a fixture that mocks the recorder,
+    in the tests that need to test the recorder.
+    """
```

Comment:
The name and docstring is a bit misleading.
The fixture should do whatever patching needs to be in place until the hass is stopped and the hass fixture has been torn down, and it may optionally return a coroutine which will be called before the HomeAssistant instance is yielded by the hass fixture.

Reply:
`patch_recorder_before_hass`?